### ISSUE TITLE
Not preserving whitespace between adjacent spans

### ISSUE THREAD

**Issue by khemarato** [wilsonzlin/minify-html#207]
If I have some HTML such as:

```html
<body>
     <span>Hello</span>
     <span>World</span>
</body>
```

This is rendered as `Hello World` in the browser as the newline + tab between the `spans` is collapsed to a single space.  Unfortunately, minhtml is minifying this down to:

```html
<body><span>Hello</span><span>World</span>
```

Which is rendered as `HelloWorld` (no space between the words).

Note that this is broken with all span-like elements, such as `<i>` or `a`, etc:

```html
<body>
     <a href="#a">Hello</a>
     <b>there</b>
     <a href="#b">World</a>
</body>
```

Note that "naked" text nodes are handled correctly:

```html
<body>
  <a href="#a">Hello</a>
  there
  <a href="#b">World</a>
</body>
```

yields `<body><a href=#a>Hello</a> there <a href=#b>World</a>` with the spaces kept in correctly.

**Comment by kyr0** [comment]
This needs a space added to out

https://github.com/wilsonzlin/minify-html/blob/master/minify-html/src/minify/element.rs#L101

I'm going to fork this repo anyways.. (no active maintanance, but the code is helpful)

**Comment by wilsonzlin** [comment]
Yes this is intentional, because it considers `<body>` a layout element, not content, which aligns closer to the spec and most common usage. For example, your text content should be in a `<p>` to follow semantic rules. This is mentioned in a major section in the [README](https://github.com/wilsonzlin/minify-html?tab=readme-ov-file#whitespace). Without these rules, it'd be impossible to make any sort of minification decisions (unless to add a lot of config) — with HTML, anyone can put almost anything anywhere, not helpful for minification decisions — which is why it follows the WHATWG spec and [repeatedly mentions this in the README](https://github.com/wilsonzlin/minify-html?tab=readme-ov-file#spec-compliance).

@kyr0 look at the [configuration file](https://github.com/wilsonzlin/minify-html/blob/2301223773dadce30a33b4c407d8b2524adeb5e2/minify-html-common/src/spec/tag/whitespace.rs#L139) as [mentioned and linked to in the README](https://github.com/wilsonzlin/minify-html?tab=readme-ov-file#element-types) if you wish to change this — it's much more straightforward.
